src/components/__tests__/CreateVocabulary.test.tsx

```
@@ -0,0 +1,737 @@
+// Self-contained mocks for hooks and utils, but Supabase will be globally mocked via setupTests.ts
+// vi.mock('@/integrations/supabase/client', () => { ... }); // REMOVED - Will use global mock
+vi.mock('@/lib/geminiUtils');
+
+// Mock other direct hook dependencies
+vi.mock('@/contexts/AuthContext');
+vi.mock('@/stores/languageStore');
+vi.mock('@/hooks/use-toast');
+vi.mock('@/lib/storyUtils'); // This is for generateAndSaveStory
+
+// Import after mocks
+import { supabase } from '@/integrations/supabase/client';
```

Biome (1.9.4)

[error] Shouldn't redeclare 'supabase'. Consider to delete it or rename it.

'supabase' is defined here:


(lint/suspicious/noRedeclare)